.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+#main.yml
+
+name: Main
+on: pull_request
+
+jobs:
+  checks:
+    runs-on: macOS-latest
+    steps:
+
+      - name: Project Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Compute key for CI cache
+        run: ./.github/workflows/compute-ci-cache-key.sh key.txt
+
+      - name: Setup cache
+        uses: actions/cache@v2.1.3
+        continue-on-error: true
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('key.txt') }}
+          restore-keys: ${{ runner.os }}-gradle-
```

Comment:
Did you try without the cache first? In my experience, this cache is so slow and unreliable/inconsistent it's actually quicker and less flaky to just re-build everything.

Reply:
Actually don't. I've been using the CI cache in a couple of projects without any issues. 

I'll remove the caching then, since it can easily be added later. Wdyt? :)

Reply:
Removed here https://github.com/square/leakcanary/pull/2012/commits/1596424e47f41cf6297112c44898bb8a9d5fe267